routes/user.route.js

```diff
@@ -1,42 +1,34 @@
+const router = require('express').Router();
 const { joiValidate } = require('express-joi');
 
 const User = require('../models/User');
 const userValidator = require('../models/validations/user');
-const { isLogin, isAdmin, isManager } = require('../middlewares/roleManager.mw');
+const { isLogin, isAdmin, isAdminOrSelf } = require('../middlewares/roleManager.mw');
 
-module.exports = app => {
-  // User Routes
-  app.get('/api/users/:id', isLogin, joiValidate(userValidator.get), async (req, res) => {
+// User Routes
+router
+  .use('/:id', isLogin)
+  .route('/:id')
+  .get(joiValidate(userValidator.get), isAdminOrSelf, async (req, res) => {
     let { id } = req.items;
     const user = await User.findById(id);
-    if (user.role === 'admin' && req.user.role !== 'admin') {
-      return res.status(403).send({
-        error: 'Access to admin details not allowed'
-      });
-    }
     res.send(user);
+  })
+  .put(joiValidate(userValidator.update), isAdminOrSelf, async (req, res) => {
+    const { id } = req.items;
+    const user = await User.findOneAndUpdate({ _id: id }, { $set: req.body }, { new: true });
```

Comment:
This we need to consider here.. if he enter `{"role": "admin"}`, It will update the role as we are not having a check for verifying whether he is a user or admin.

Reply:
Ok, I have a solution, on line 19 if he is not admin, we will delete the role key and continue

Reply:
Creating an issue for this

Reply:
Fixed in #55